x-pack/test/defend_workflows_cypress/agent.ts

```diff
@@ -0,0 +1,182 @@
+
+import os from 'node:os';
+import execa from 'execa';
+import { find } from 'lodash';
+import { ToolingLog } from '@kbn/tooling-log';
+import axios, { AxiosRequestConfig } from 'axios';
+import { ChildProcess } from 'child_process';
+// import { getLatestVersion } from './artifact_manager';
+import { Manager } from './resource_manager';
+
+const VM_NAME = 'endpoint10';
+
+export interface AgentManagerParams {
+  user: string;
+  password: string;
+  kibanaUrl: string;
+  esHost: string;
+  esPort: string;
+}
+
+export class AgentManager extends Manager {
+  private params: AgentManagerParams;
+  private log: ToolingLog;
+  private agentProcess?: ChildProcess;
+  private requestOptions: AxiosRequestConfig;
+  private agentPolicyId?: string;
+  constructor(params: AgentManagerParams, log: ToolingLog, requestOptions: AxiosRequestConfig) {
+    super();
+    this.log = log;
+    this.params = params;
+    this.requestOptions = requestOptions;
+    this.agentPolicyId = undefined;
+  }
+
+  public async setup() {
+    this.log.info('Running agent preconfig');
+
+    const {
+      data: { items: packages },
+    } = await axios.post(
+      `${this.params.kibanaUrl}/api/fleet/epm/packages/_bulk`,
+      {
+        packages: ['endpoint'],
+      },
+      this.requestOptions
+    );
+
+    const {
+      data: {
+        item: { id: agentPolicyId },
+      },
+    } = await axios.post(
+      `${this.params.kibanaUrl}/api/fleet/agent_policies?sys_monitoring=true`,
+      {
+        name: 'endpoint',
+        description: '',
+        namespace: 'default',
+        monitoring_enabled: ['logs', 'metrics'],
+      },
+      this.requestOptions
+    );
+
+    await axios.post(
+      `${this.params.kibanaUrl}/api/fleet/package_policies`,
+      {
+        name: 'endpoint',
+        description: '',
+        namespace: 'default',
+        policy_id: agentPolicyId,
+        enabled: true,
+        inputs: [
+          {
+            enabled: true,
+            streams: [],
+            type: 'ENDPOINT_INTEGRATION_CONFIG',
+            config: {
+              _config: {
+                value: {
+                  type: 'endpoint',
+                  endpointConfig: {
+                    preset: 'NGAV',
+                  },
+                },
+              },
+            },
+          },
+        ],
+        package: {
+          name: 'endpoint',
+          title: 'Elastic Defend',
+          version: packages[0].version,
+        },
+      },
+      this.requestOptions
+    );
+
+    this.agentPolicyId = agentPolicyId;
+
+    this.log.info('Getting agent enrollment key');
+    const apiKeys = await axios.get(
+      `${this.params.kibanaUrl}/api/fleet/enrollment_api_keys?page=1&perPage=10000`,
+      this.requestOptions
+    );
+    const enrollmentToken = find(
+      apiKeys.data.items,
+      (item) => item.policy_id === agentPolicyId
+    ).api_key;
+
+    const hostIp = find(os.networkInterfaces().en0, { family: 'IPv4' })?.address ?? '0.0.0.0';
```

Comment:
@paul-tavares here is an example using `os` module

Reply:
🤔 - this is looking at `en0` only. There could be multiple network interfaces, are you sure that `en0` will always be the correct one?

Reply:
on second thought - after reading up on `en0` again to refresh the 🧠 - yes seems like a good solution 👍 

Will you make the change to the `common` utility under scripts along with this PR? or should I include it in a subsequent PR?

Reply:
`en0` seems most common to :apple: machines, and wifi.  Here is what I get:

```
> Object.keys(os.networkInterfaces())
[ 'lo', 'enp5s0' ]
```

Reply:
Thank you @pzl, now it should be working properly in your case as well